packages/client/src/lobby/createReplayJSON.ts

```diff
@@ -50,23 +49,57 @@ export default function createJSONFromReplay(room: string) {
   game.actions = getGameActionsFromState(replay);
 
   // Add the hypothesis from log, after current segment
-  const states = replay.hypothetical!.states;
-  const log = states[states.length - 1].log;
-  if (replay.segment < log.length) {
-    const slice = log.slice(replay.segment + 1);
-    const actions = getGameActionsFromLog(slice);
-    game.actions.push(...actions);
+  if (replay.hypothetical !== null) {
+    const states = replay.hypothetical.states;
+    const log = states[states.length - 1].log;
+    if (replay.segment < log.length) {
+      const slice = log.slice(replay.segment + 1);
```

Comment:
slice is a bad name, the variable name should describe what the data actually contains

Reply:
its marked as resolved but i dont see it changed

Reply:
I haven't sent my commit yet. I'm fixing and marking as I go

Reply:
do you do it in order from top to bottom? if yes, then maybe its better to wait until the commit is live, so that i don't get confused